### ISSUE TITLE
HTSLib Support

### ISSUE THREAD

**Issue by reedacartwright** [denovogear/denovogear#73]
CMake is not able to find the version of HTSLib in many cases.  We need to link against htslib to extract the version information during the build phase.  Note: this will not work when cross compiling.

**Comment by zmertens** [comment]
I wanted to try to build DenovoGear on my machine (Linux Mint 17.2), and I ran into this CMake warning and I was able to resolve it by [building HTSlib 1.3.1 from source](https://github.com/samtools/htslib/tree/master). 

Additionally, there might be an issue with the Debian-based package manager on Linux Mint. First, I had downloaded HTSlib using `apt-get` and it said the package in the repositories was `0.2.0` -- [note the official repo](http://packages.ubuntu.com/trusty/libhts0). When I tried to build DNG the compiler complained about missing headers such as `<htslib/hfile.h>` so I assumed it was definitely the wrong version. [This](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=822701) was the only issue I could find in the Debian packages but seems like it might be related somehow.

At any rate, building HTSlib from source seemed to solve this problem for me.

-- Zach Mertens
